test/use-checkout-pricing.test.js

```diff
@@ -1,308 +1,119 @@
+import useCheckoutPricing, { throwError } from '../lib/use-checkout-pricing';
 import { withElements } from './support/helpers';
-import Promise from "promise";
+import { server } from 'recurly.js/test/server/';
 
-const subscriptionPricingReturn = {
-  addon: jest.fn(() => subscriptionPricingReturn),
-  catch: jest.fn(() => subscriptionPricingReturn),
-  done: jest.fn(() => subscriptionPricingReturn),
-  plan: jest.fn(() => subscriptionPricingReturn),
-  tax: jest.fn(() => subscriptionPricingReturn),
-  currency: jest.fn(() => subscriptionPricingReturn)
-};
-
-const PRICING_METHODS = [
-  'reset',
-  'remove',
-  'reprice',
-  'addon',
-  'address',
-  'coupon',
-  'currency',
-  'giftCard',
-  'plan',
-  'shippingAddress',
-  'tax'
-];
-
-const checkoutPricingReturn = {
-  address: jest.fn(() => checkoutPricingReturn),
-  adjustment: jest.fn(() => checkoutPricingReturn),
-  catch: jest.fn(() => checkoutPricingReturn),
-  coupon: jest.fn(() => checkoutPricingReturn),
-  currency: jest.fn(() => checkoutPricingReturn),
-  done: jest.fn(cb => {
-    cb();
-    return checkoutPricingReturn;
-  }),
-  giftCard: jest.fn(() => checkoutPricingReturn),
-  reprice: jest.fn(() => checkoutPricingReturn),
-  shippingAddress: jest.fn(() => checkoutPricingReturn),
-  subscription: jest.fn(() => checkoutPricingReturn),
-  tax: jest.fn(() => checkoutPricingReturn),
-  pricing: { PRICING_METHODS }
-};
-
-describe('useCheckoutPricing', function() {
+describe('useCheckoutPricing', function () {
   const { Recurly } = window.recurly;
 
   beforeEach(() => {
     jest.clearAllMocks();
-    window.recurly.Recurly = () => {
-      return {
-        ...new Recurly('Public key'),
-        Pricing: {
-          Subscription: jest.fn(() => subscriptionPricingReturn),
-          Checkout: jest.fn(() => checkoutPricingReturn)
-        },
-      };
-    };
+    window.recurly.Recurly = () => new Recurly({
+      publicKey: 'Public key',
+      api: `http://localhost:${process.env.PORT || 9877}`
+    });
   });
 
-  afterEach(() => {
-    window.recurly.Recurly = Recurly;
+  afterAll(() => {
+    server.close();
   });
 
   describe('loading', () => {
     it('should be true initially, then false after initial pricingPromise resolves', async () => {
       const initialInput = { subscriptions: [{ plan: 'basic', quantity: 2 }] };
       const { result, waitForNextUpdate } = renderUseCheckoutPricing(initialInput);
 
-      await act(async () => {
-        expect(result.current[0].loading).toBe(true);
-        await waitForNextUpdate();
-        await expect(result.current[0].loading).toBe(false);
-      });
-    });
-  });
-
-  describe('subscriptions', () => {
-    it('should call subscriptionPricing.plan for each subscription', async () => {
-      const initialInput = {
-        subscriptions: [
-          { plan: 'basic', quantity: 2 },
-          { plan: 'advanced', quantity: 5 }
-        ]
-      };
+      await expect(result.current[0].loading).toBe(true);
```

Comment:
This can't be abstracted to a variable for reuse because `.current` is a getter.